src/mouse_events.cpp

```diff
@@ -785,17 +787,23 @@ void mouse_handler::select_or_action(bool browse)
 	wb::future_map_if_active planned_unit_map;
 
 	if(game_lua_kernel* lk = pc_.gamestate().lua_kernel_.get()) {
-		lk->select_hex_callback(last_hex_);
+		lk->select_hex_callback(clicked_hex);
+		// The callback will typically modify clicked_hex, so make sure it's still valid
```

Comment:
Comment needs an update.

Reply:
In fact, it still does modify `clicked_hex`, because I took the quickest route to changing the return value of the callback – it just sets the location to an invalid one if the callback returns true.

I'll still modify it, because I'm not sure that's a "typical" use-case.